baremetal/metal3machine_manager_test.go

```diff
@@ -525,28 +544,38 @@ var _ = Describe("Metal3Machine manager", func() {
 				Namespace: namespaceName,
 				Labels:    map[string]string{"key1": "value1"},
 			},
+			Status: bmh.BareMetalHostStatus{
+				Provisioning: bmh.ProvisionStatus{
+					State: bmh.StateAvailable,
+				},
+			},
 		}
 		hostWithUnhealthyAnnotation := bmh.BareMetalHost{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:        "hostWithUnhealthyAnnotation",
 				Namespace:   namespaceName,
 				Annotations: map[string]string{capm3.UnhealthyAnnotation: "unhealthy"},
 			},
+			Status: bmh.BareMetalHostStatus{
+				Provisioning: bmh.ProvisionStatus{
+					State: bmh.StateAvailable,
+				},
+			},
 		}
 		hostWithNodeReuseLabelSetToKCP := bmh.BareMetalHost{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      "hostWithNodeReuseLabelSetToKCP",
 				Namespace: namespaceName,
-				Labels:    map[string]string{nodeReuseLabelName: "kcp-pool1"},
+				Labels:    map[string]string{"key1": "value1", nodeReuseLabelName: "kcp-test1"},
 			},
-		}
-		hostWithNodeReuseLabelSetToMD := bmh.BareMetalHost{
```

Comment:
This host `hostWithNodeReuseLabelSetToMD` meant to test the MD use case, wonder why we are removing it at all

Reply:
Testing `hostWithNodeReuseLabelSetToMD ` requires MachineSet to be passed to the tests. In this case, we need further expand test. For this reason I decided to remove this test case